### PR TITLE
FindSpace with options

### PR DESCRIPTION
**Pull request by hugelgupf**
* Improved FindSpace API to locate holes in MemoryMap.

Depends on https://github.com/u-root/u-root/pull/2878

### CODE REVIEW

**Comment by rminnich** [comment]
quick question: could this package be of any use: https://pkg.go.dev/github.com/go-camp/interval

**Comment by hugelgupf** [comment]
> quick question: could this package be of any use: https://pkg.go.dev/github.com/go-camp/interval

I think we have the primitives we want here with `kexec.Range`, `kexec.TypedRange`, and `kexec.MemoryMap`. The new things being added here are not available in that package (like "find a whole in these intervals that is aligned")

**Comment by hugelgupf** [comment]
Ready for review.

File: pkg/boot/kexec/memory_linux.go [hunk 1]
```diff
@@ -21,13 +21,7 @@ var pageMask = uint(os.Getpagesize() - 1)
...
 // ErrNotEnoughSpace is returned by the FindSpace family of functions if no
 // range is large enough to accommodate the request.
-type ErrNotEnoughSpace struct {
-	Size uint
-}
-
-func (e ErrNotEnoughSpace) Error() string {
-	return fmt.Sprintf("not enough space to allocate %#x bytes", e.Size)
-}
+var ErrNotEnoughSpace = fmt.Errorf("not enough space to allocate bytes")
```
Comment (rminnich) [inline pkg/boot/kexec/memory_linux.go:24]: Every time I see these I wonder about using syscall.ENOMEM :-)

**Review by rminnich — APPROVED** [review]

**Pull request merged**